scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -55,15 +55,15 @@ abstract class ScanResultsStorage : PackageBasedScanStorage {
     open val name: String = javaClass.simpleName
 
     /**
-     * Return all [ScanResult]s contained in this [ScanResultsStorage] corresponding to the package denoted by the given
-     * [id] wrapped in a [Result].
+     * Return all [ScanResult]s contained in this [ScanResultsStorage] corresponding to the [package][pkg] wrapped in a
+     * [Result].
      */
-    fun read(id: Identifier): Result<List<ScanResult>> {
-        val (result, duration) = measureTimedValue { readInternal(id) }
+    fun read(pkg: Package): Result<List<ScanResult>> {
```

Comment:
commit and this read() function: I have difficulties understanding which parts of the package will be used to determine which scan results are being returned and how. Can you help?

Reply:
It's indeed a bit hard to understand as the relevant parts are not implemented yet... because they couldn't, as the ClearlyDefined scan storage provider did not have access to the necessary information before this change.

Basically, the ClearlyDefined implementation would need to have access to the VCS location, source artifact location (and maybe also the binary source artifact location) in order to create a proper [Coordinate](https://github.com/oss-review-toolkit/ort/blob/59a4404f1f01434336635e6c6495f407bd6108a0/clients/clearly-defined/src/main/kotlin/Curation.kt#L103) for querying harvested (scan) results. A *Coordinate* includes a [ComponentType](https://github.com/oss-review-toolkit/ort/blob/681df5e2e5392d11901bfa6d40c0cbc821fe5005/clients/clearly-defined/src/main/kotlin/Enums.kt#L31) and [Provider](https://github.com/oss-review-toolkit/ort/blob/681df5e2e5392d11901bfa6d40c0cbc821fe5005/clients/clearly-defined/src/main/kotlin/Enums.kt#L60), and in order to determine these properly, more *Package* properties than just the *Identifier* are required.

Reply:
As `Package` contains many irrelevant properties I would prefer if the interface would only require the relevant values to have a cleaner API. But I would be fine with looking into it after fixing the ClearlyDefinedStorage.